### ISSUE TITLE
[UNIQUE] 1.17 support

### ISSUE THREAD

**Issue by AmauryCarrade** [zDevelopers/ImageOnMap#198]
This is a ticket for every request related to 1.17 support.

---

- [ ] **Splatter maps not working correctly**
     Whenever i make a make using /tomap it creates it fine but if it is over 1x1 and i try to put it on a item frame it doesnt fill all of them and when i take it back it disappears. This only started happening after i upgraded to 1.17
     - #192
     - #194
     - #195

- [ ] **Rendering looks slow**
     - #195
- [ ] **Cannot create maps**
    Map rendering failed: `java.lang.NullPointerException: Cannot read field "mapView" because "worldmap" is null`
     - #199
- [ ] **Unable to get splatter map attribute on item** (because NMS classes are not found)
     - #200 (trace)
- [ ] **Unable to rename map** (because NMS classes are not found)
     - #201 
 - [ ] **After restarting the server, the maps dont show images**
     - #203 (no details)
     - #210 (More details provided)
 
---

Add your 1.17 compatibility requests in this issue. Thanks!

**Comment by Vorald** [comment]
Sorry for not writting in the right place, do you know a way to the fix my problem? 
Thanks for the quick response before!

**Comment by AmauryCarrade** [comment]
First of all, try to use the latest Spigot/Paper possible. Some users reported improvements with the latest builds. But we have to update ImageOnMap itself because some internals changed, so for your problem, you'll _probably_ have to wait until we release a 1.17-compatible version. Work is in progress but it's on our free time so it's… when possible :) .

And, you're welcome!

**Comment by Vorald** [comment]
> First of all, try to use the latest Spigot/Paper possible. Some users reported improvements with the latest builds. But we have to update ImageOnMap itself because some internals changed, so for your problem, you'll _probably_ have to wait until we release a 1.17-compatible version. Work is in progress but it's on our free time so it's… when possible :) .
> 
> And, you're welcome!

Thanks!

**Comment by Zoro-6191** [comment]
waiting.. patiently.. :(

**Comment by dimethro** [comment]
me too when wil the 1.17 come out will it work if we render the source on java 16?

**Comment by Zoro-6191** [comment]
zDev I apologize for this, since you don't supposed to be working on the plugin, I'll leave a link to a [separate plugin](https://www.spigotmc.org/resources/yamipa.88709/) for people who can't wait for it to come out

**Comment by Aurelien30000** [comment]
Any update here?

**Comment by Vlammar** [comment]
In progress but I still have some anoying bugs to fix, such as this one https://cdn.discordapp.com/attachments/722920026574684211/860821198035877918/unknown.png.

**Comment by Aqueuse** [comment]
Hello, Is ImageOnMap mostly ready for 1.17 ? I would like to use it and I would be happy to help if needed

**Comment by Vlammar** [comment]
To have an idea on what's left see #219 and https://github.com/zDevelopers/QuartzLib/pull/80

**Comment by Vlammar** [comment]
Waiting for the PR review. once done I will lock this and subsequent issues

**Comment by srnyx** [comment]
@Vlammar I noticed you merged https://github.com/zDevelopers/ImageOnMap/pull/219, is there anything else that needs to be done?

**Comment by Vlammar** [comment]
@srnyx @Aqueuse https://github.com/zDevelopers/ImageOnMap/releases/tag/v4.2.0 Still waiting for the final release but here is an actual pre release from the lastest indev build. If you want to build it you have to use the update/1.17 branch of quartzlib and change the pom.xml of IoM to make sure you use "0.0.5-SNAPSHOT" of QL. Don't hesitate to report me any issues you may find

**Comment by srnyx** [comment]
> use the update/1.17 branch of quartzlib

How do I do this? I can't seem to build it.

> change the pom.xml of IoM to make sure you use "0.0.5-SNAPSHOT" of QL.

How do I do this as well? Is it not already changed in the pre-release JAR?

**Comment by Vlammar** [comment]
You can use the IoM4.2.0.jar directly. I was only giving the steps to build it. We haven't release yet QL 0.0.5 that's why if you try to build IoM 4.2.0 this won't works

**Comment by srnyx** [comment]
> You can use the IoM4.2.0.jar directly. I was only giving the steps to build it. We haven't release yet QL 0.0.5 that's why if you try to build IoM 4.2.0 this won't works

Ohhhhhh! Alright, thank you 😄

**Comment by srnyx** [comment]
> Don't hesitate to report me any issues you may find

![image](https://user-images.githubusercontent.com/25808801/142491668-ab0b0776-0e60-465b-99c8-ef3110826235.png)

**Comment by Vlammar** [comment]
You have any output in the console ? @srnyx 
What minecraft version are you running ? (and paper/spigot?)

**Comment by srnyx** [comment]
> You have any output in the console ?

No

> What minecraft version are you running ? (and paper/spigot?)

Purpur 1.17.1

**Comment by srnyx** [comment]
@Vlammar images don't work at all: https://paste.srnyx.xyz/zovoqoduwe.md
The files aren't being generated in the `maps` folder.
*When I use the command, I get the image, I can place it in an item frame, but when I remove them from the item frame and then place it back it, it gets deleted.*

**Comment by Vlammar** [comment]
@srnyx don't know how I have done it but the version uploaded was done with old indev code despite being builded after the merge with #219 . That should work fine right now

**Comment by brysondev** [comment]
> @Vlammar images don't work at all: https://paste.srnyx.xyz/zovoqoduwe.md The files aren't being generated in the `maps` folder. _When I use the command, I get the image, I can place it in an item frame, but when I remove them from the item frame and then place it back it, it gets deleted._

@Vlammar the `images` folder isn't being created, hence the error he's experiencing. If you manually create a `images` folder in the directory, the plugin works fine.

**Comment by srnyx** [comment]
> > @Vlammar images don't work at all: [paste.srnyx.xyz/zovoqoduwe.md](https://paste.srnyx.xyz/zovoqoduwe.md) The files aren't being generated in the `maps` folder. _When I use the command, I get the image, I can place it in an item frame, but when I remove them from the item frame and then place it back it, it gets deleted._
> 
> @Vlammar the `images` folder isn't being created, hence the error he's experiencing. If you manually create a `images` folder in the directory, the plugin works fine.

I think it's called `maps`, which was being generated, just the image files themselves weren't. Updating the JAR file fixed it.

**Comment by srnyx** [comment]
> I think it's called `maps`, which was being generated, just the image files themselves weren't. Updating the JAR file fixed it.

Scratch this, @brysondev was right, the `images` folder isn't being generated. Manually creating the `images` folder fixes it.

**Comment by srnyx** [comment]
Now we need 1.18 support 😔

**Comment by plopilpy** [comment]
Is there any solution for splatter maps not working? All I see are the threads that point to here but this doesn't have a definite answer

**Comment by brysondev** [comment]
> Is there any solution for splatter maps not working? All I see are the threads that point to here but this doesn't have a definite answer

You need to be more specific about your issue. As the thread suggests they are working on fixing the issues stated above. What error are you getting?

**Comment by plopilpy** [comment]
I'm not getting any errors. I follow the steps exactly as described, and it doesn't work.

**Comment by Vlammar** [comment]
> I'm not getting any errors. I follow the steps exactly as described, and it doesn't work.

Maybe because we are working on it and there is a pre release  https://github.com/zDevelopers/ImageOnMap/releases/tag/v4.2.0

**Comment by Vlammar** [comment]
1.17 support is finally here with 1.18 suport. see the 4.2.1 of IoM https://github.com/zDevelopers/ImageOnMap/releases/tag/v4.2.1

**Comment by srnyx** [comment]
> 1.17 support is finally here with 1.18 suport. see the 4.2.1 of IoM [`v4.2.1` (release)](https://github.com/zDevelopers/ImageOnMap/releases/tag/v4.2.1)

I assume that means this issue can be closed now, right?

**Comment by Aurelien30000** [comment]
Hey, suffer a breaking error on 1.17.1, could you help me? Thanks! https://paste.domicraft.fr/ugujerapat.properties

**Comment by Vlammar** [comment]
> I assume that means this issue can be closed now, right?

Yes I'm waiting for the 4.2.1 official release before closing this one

**Comment by Vlammar** [comment]
> Hey, suffer a breaking error on 1.17.1, could you help me? Thanks! https://paste.domicraft.fr/ugujerapat.properties

4.2.0 should fix this (4.2.1 even works with 1.18)

**Comment by Aurelien30000** [comment]
I'm using ImageOnMap version 4.2.1 🤔

**Comment by Vlammar** [comment]
Same issue with 4.2.0?

**Comment by Vlammar** [comment]
> I'm using ImageOnMap version 4.2.1 🤔

Same issue with 4.2.0 ?

**Comment by Aurelien30000** [comment]
> > I'm using ImageOnMap version 4.2.1 🤔
> 
> Same issue with 4.2.0 ?

Sorry for late answer, 4.2.0 doesn't support 1.18 iirc, so nothing interesting here.

**Comment by MistakingManx** [comment]
1.21.7 doesn't load.
